### PR TITLE
Add yarn lock file and additional info in docs

### LINKED ISSUE #602: Use Yarn Package Manager
[Yarn](https://yarnpkg.com/) - a new package manager that extends npm was just released. It looks like a big improvement over npm. We should give it a shot!
## References
- [Yarn](https://yarnpkg.com/)
- [Yarn's Github](https://github.com/yarnpkg/)

### PR DESCRIPTION
**Pull request by sashashakun**
# What's in this PR?

Just add yarn.lock and additional info in docs
## References

Fixes #602 

Progress on: 100%

### CODE REVIEW

**Comment by JoshSmith** [comment]
@sashashakun does Yarn not handle `bower install`?

Do we need to anything in our CI to use this as opposed to `npm` and `bower` directly?

**Comment by JoshSmith** [comment]
@sashashakun I would get rid of that "Merge branch 'develop'" commit. Much prefer `rebase` instead of `merge`.

**Comment by JoshSmith** [comment]
https://github.com/code-corps/code-corps-ember/blob/develop/docs/SQUASHING.md#what-is-a-rebase

**Comment by sashashakun** [comment]
>does Yarn not handle bower install?
As I see think currently yarn have some problems with bower but also have movement to support and handle it: https://github.com/yarnpkg/yarn/pull/896
>Do we need to anything in our CI to use this as opposed to npm and bower directly?
Because of issue, I mentioned above, I think you should use yarn with bower separately

Also, will fix merge commit, sorry for this)

**Comment by JoshSmith** [comment]
@sashashakun if you look at our `circle.yml` that might give you some indication for how CI runs currently, along with any output you can see from the build here in GitHub.

Let me know if I'm being unclear, and always feel free to stop by #ember channel in [our Slack](http://slack.codecorps.org)

File: circle.yml [hunk 1]
```diff
@@ -8,8 +8,9 @@ dependencies:
     - node_modules
     - bower_components
   override:
-    - npm install -g bower
-    - npm install
+    - curl -o- -L https://yarnpkg.com/install.sh | bash
```
Comment (JoshSmith) [inline circle.yml]: We might want to do this in a separate pre-dependency step.

**Comment by sashashakun** [comment]
At the moment I need to go, but will continue my work on this in few hours. 
Will configure CI at the correct way.

**Comment by sashashakun** [comment]
@JoshSmith Hello again! I apologize for the inconvenience, I move yarn installation to `pre` step in CI config, but as you can see, there is some problem with phantomjs. May be I miss something in docs?

**Comment by JoshSmith** [comment]
@sashashakun you're not the only one having Phantom issues. @begedin and @rileytaylor are having similar problems.

I need to dig in and figure out what change was introduced that caused this and why.

**Comment by JoshSmith** [comment]
@sashashakun I'm about to merge #613 which should fix these issues. Once it's merged please rebase and see if it's fixed.

**Comment by JoshSmith** [comment]
@sashashakun would you mind attempting to rebase now?

**Comment by sashashakun** [comment]
@JoshSmith Sorry for the late response. I made a rebase, but unfortunately problem is still here. Also I can't see any useful output after falling.🤔

**Comment by JoshSmith** [comment]
@sashashakun is this similar to #618?

**Comment by sashashakun** [comment]
As I can see on the output of CI, no(

**Comment by JoshSmith** [comment]
@sashashakun it looks like Circle's docs for this have changed (https://circleci.com/docs/install-and-use-yarn/). Also, whatever was causing failures for PRs before is gone.

Can you fetch latest and see if you can push up again? Thanks!

**Comment by JoshSmith** [comment]
After this issue tonight https://github.com/visionmedia/debug/issues/347 we definitely need to prioritize this. Demonstrates some preventable time lost by simply having a `yarn.lock`.

**Comment by sashashakun** [comment]
@JoshSmith thanks for the link, will make improvements soon👌

**Comment by JoshSmith** [comment]
Closing in favor of #975 so can take over.

**Pull request closed without merging**